### ISSUE TITLE
Unhelpful UnsatisfiedRequirement error message

### ISSUE THREAD

**Issue by eugenhu** [alecthomas/injector#129]
Whenever there is an unsatisfied requirement, I see the message,
```
injector.UnsatisfiedRequirement: <exception str() failed>
```
I am using Python 3.5.7.

I think the cause of this is
https://github.com/alecthomas/injector/blob/a461f722adcf8e23f0fdd4e122f101029a74877c/injector/__init__.py#L585
which creates a new `UnsatisfiedRequirement` object with only one argument, but the `__str__()` function
https://github.com/alecthomas/injector/blob/a461f722adcf8e23f0fdd4e122f101029a74877c/injector/__init__.py#L181-L186
utilises `self.args[0]` and `self.args[1]`, which causes an exception.

I'm not sure what `self.args[0]` and `self.args[1]` are meant to represent but modifying line 585 to
```python
raise UnsatisfiedRequirement(None, interface)
```
seems to output a sensible message when the error is thrown.

**Comment by jstasiak** [comment]
Thanks!

**Comment by jstasiak** [comment]
The fix for this has just been released as part of the version 0.18.1.

**Comment by eugenhu** [comment]
Oh sorry, I've just realised I forgot to mention that I also had to change line 186

https://github.com/alecthomas/injector/blob/a461f722adcf8e23f0fdd4e122f101029a74877c/injector/__init__.py#L186

to

```python
return '%sunsatisfied requirement on %s' % (on, _describe(self.args[1])) 
```

since I was getting an `AttributeError` that `self.args[1]` doesn't have an attribute `interface`. I don't really know what `self.args[1]` is supposed to be so I don't know if this is the right way to fix it.

My bad.

---

Short example:
```
from injector import Injector, inject

@inject
def main(password: str):
    print(password)

injector = Injector(auto_bind=False)
injector.call_with_injection(main)
```
